rkt/api_service.go

```diff
@@ -289,26 +283,27 @@ func getNetworks(p *pod) []*v1alpha.Network {
 func getBasicPod(p *pod) (*v1alpha.Pod, *schema.PodManifest, error) {
```

Comment:
The doc-comment should reflect that a `POD_STATE_UNDEFINED` can be returned I think.

Do we want to distinguish failures with a little more granularity here though? I think there might be room for distinguishing broken pod (`Id` present) with absent pod (maybe a final bool argument?).
With this change, if you pass an Id into the api it will always echo it back, yes? Do you actually want a non-existent Id to echo back?